pinot-broker/src/main/java/org/apache/pinot/broker/broker/ServiceAutoDiscoveryFeature.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.broker.broker;
+
+import java.io.IOException;
+import javax.inject.Inject;
+import javax.ws.rs.core.Feature;
+import javax.ws.rs.core.FeatureContext;
+import org.glassfish.hk2.api.DynamicConfigurationService;
+import org.glassfish.hk2.api.MultiException;
+import org.glassfish.hk2.api.Populator;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.hk2.utilities.ClasspathDescriptorFileFinder;
+import org.glassfish.hk2.utilities.DuplicatePostProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Auto scan and discovery classes annotated with Service tags. This enables the feature similar to Spring's
+ * auto-discovery, that if a class is annotated with some tags like @Service, the class will be auto-loaded
+ * The code are mostly from https://mkyong.com/webservices/jax-rs/jersey-and-hk2-dependency-injection-auto-scanning/
+ * <p>
+ * To make a class auto-loaded, what we need to do is to add the below dependency into project:
+ * <p>
+ * <pre>
+ *      &lt;dependency&gt;
```

Comment:
If I use <pre>{@code .... } </pre>, our check-style plugin will complain and fail the build.
Here I was forced to use HTML encode